src/ai/backend/manager/reporters/smtp.py

```diff
@@ -0,0 +1,137 @@
+import asyncio
+import enum
+import logging
+from abc import abstractmethod
+from dataclasses import dataclass
+from datetime import datetime
+from email.mime.text import MIMEText
+from typing import Final
+
+import aiosmtplib
+
+from ai.backend.logging.utils import BraceStyleAdapter
+from ai.backend.manager.models.audit_log import OperationStatus
+from ai.backend.manager.reporters.types import (
+    AbstractReporter,
+    FinishedActionMessage,
+    StartedActionMessage,
+)
+
+log = BraceStyleAdapter(logging.getLogger(__spec__.name))
+
+
+UNKNOWN_ENTITY_ID: Final[str] = "(unknown)"
+
+
+class SMTPTriggerPolicy(enum.Flag):
+    PRE_ACTION = enum.auto()
+    POST_ACTION = enum.auto()
+    ON_ERROR = enum.auto()
```

Comment:
could you please apply it as a StrEnum to reduce mistakes?

Reply:
I think using `enum.Flag` provides a more convenient way to express policies for when to log events, compared to using a `list[StrEnum]`.